### ISSUE TITLE
[BUG] Error `cause` not being respected

### ISSUE THREAD

**Issue by mmkal** [taiga-family/micro-sentry#17]
# 🐞 Bug report

### Description

We are calling `Sentry.report(new Error('Something bad happened'), {cause: 'Some useful context about the bad thing'})` on a micro-sentry client, but we only see the "Something bad happened" message in sentry.io. I would expect the cause to appear under the "extras" section in the Sentry UI. It's not in the JSON payload sent to sentry a all though

### Reproduction

I don't know that it's possible to reproduce in stackblitz since it's about what gets sent to sentry, but it should be easy to reproduce if you have a local sentry set up. Just create a new `@micro-sentry/browser` client and call the code snippet above.

### Expected behavior

Include `cause` ([MDN docs](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Error/cause))

### Versions

- Package [e.g. @micro-sentry/browser@6.0.3]

If needed:

- Angular/React/Vue [e.g. Angular 11]

**Comment by mmkal** [comment]
For anyone else who gets this, here's a workaround: just extend the class, so you can use the protected `send` method:

```ts
import {BrowserMicroSentryClient as Base} from '@micro-sentry/browser'

class BrowserMicroSentryClient extends Base {
  report(error: Error) {
    const prepared = this.prepare(error)
    prepared.extra ||= {}
    prepared.extra.cause ||= error.cause
    Object.assign(prepared.extra, error)
    this.send(prepared)
  }
}
```

(the Object.assign adds anything else hanging off the Error object - what the official Sentry client does too)

**Comment by mmkal** [comment]
I opened https://github.com/taiga-family/micro-sentry/pull/18 with the workaround we're using, but it probably isn't up to scratch/mergeable - I haven't looked into docs/tests etc. But feel free to push to that branch - I don't think I'll be able to dive in any time soon sadly.

**Comment by sevaru** [comment]
Thanks!

I've made a [comment](https://github.com/taiga-family/micro-sentry/pull/18#issuecomment-1932216721) on the PR.

TL;DR:
Extending the class as your workaround is indeed the best approach for now!
